lib/cext/preprocess.rb

```diff
@@ -783,6 +783,103 @@ def tuple_new_patch(ctx, slf)
         replacement: 'if (rb_tr_obj_equal(Qtrue, obj))'
       }
     ]
+  },
+  'database.c' => {
+    gem: 'sqlite3',
+    patches: [
+      { # tracefunc
+        match: 'VALUE self = (VALUE)data;',
+        replacement: 'VALUE self = rb_tr_managed_from_handle(data);'
+      },
+      { # trace
+        match: 'sqlite3_trace(ctx->db, NIL_P(block) ? NULL : tracefunc, (void *)self);',
+        replacement: 'sqlite3_trace(ctx->db, NIL_P(block) ? NULL : tracefunc, rb_tr_handle_for_managed_leaking(self));'
+      },
+      { # rb_sqlite3_busy_handler
+        match: 'VALUE self = (VALUE)(ctx);',
+        replacement: 'VALUE self = rb_tr_managed_from_handle(ctx);'
+      },
+      { # busy_handler
+        match: 'ctx->db, NIL_P(block) ? NULL : rb_sqlite3_busy_handler, (void *)self);',
+        replacement: 'ctx->db, NIL_P(block) ? NULL : rb_sqlite3_busy_handler, rb_tr_handle_for_managed_leaking(self));'
+      },
+      { # rb_sqlite3_func
+        match: 'VALUE callable = (VALUE)sqlite3_user_data(ctx);',
+        replacement: 'VALUE callable = rb_tr_managed_from_handle(sqlite3_user_data(ctx));'
+      },
+      { # define_function_with_flags
+        match: 'NUM2INT(flags),',
+        replacement: 'flags,'
```

Comment:
Why is this needed?

Reply:
Evidently it's not. In a fit of debugging I thought it helped make progress, but I just removed it and everything seems to work.